charactersheet/charactersheet/services/persistence_service_components/persistence_service_predicates.js

```diff
@@ -3,17 +3,28 @@
 /**
  * A convenience class for constructing multi-property queries for
  * the persistence service.
- * 
+ *
  * @field key  name of property that will be compared
  * @field value  desired value to be compared against
  */
 function KeyValuePredicate(key, value) {
     var self = this;
-    
+
     self.key = key;
     self.value = value;
 
     self.matches = function(element){
         return element[self.key] === self.value;
     };
-}
\ No newline at end of file
+}
+
+
+function NotPredicate(predicate) {
```

Comment:
Really liking the predicates. They make complex queries really easy and readable without lots of imperative logic.